docs/src/pages/styles/advanced/ThemeNesting.tsx

```diff
@@ -1,5 +1,6 @@
 import * as React from 'react';
-import { ThemeProvider, makeStyles, createStyles } from '@material-ui/core/styles';
+import { createStyles } from '@material-ui/styles';
```

Comment:
Let's just remove every internal usage of `createStyles` and replace it with `as const` assertions where necessary. Makes things easier for us.